### PR TITLE
Disable showing money effect when paused

### PR DESCRIPTION
**Pull request by KatieZeldaKat**
Previously when building in pause mode, money effects would clutter the screen and not go away unless a player resumed the game for the animation to take place. This pull request makes it so that the money effect is no longer created when the game is paused. This is also a re-implementation of former functionality, as this used to be the case after v0.1.0 and before v0.2.3 (more info below).

---

In #5733, the money effect was disabled when building in pause mode to prevent a crash when too many sprites were being animated at once (#3479). This used to be implemented as such:

```
// Create a +/- money text effect
if (cost != 0 && game_is_not_paused())
    money_effect_create(cost);
```

While this code still existed in v0.2.3, for whatever reason, the money effect started to be created even when the game was paused. Multiple users have since mentioned that it would be beneficial to not have these sprites generate when paused, though some have also mentioned the alternative of animating the sprites when paused instead:

- [Reddit: How to disable money animations from building?](https://www.reddit.com/r/openrct2/comments/uy418t/how_to_disable_money_animations_from_building/)
- #17036

Through intensive digging, I don't believe this functionality was removed intentionally, especially as the changelog makes no mention of reverting this functionality in v0.2.3 after describing its introduction in v0.1.0. As such, this pull request adds back the logic of the above code block, now inside of `MoneyEffect::CreateAt()`.

Although it could be argued that the true solution to the sprite problem when building in pause mode is to animate them when paused, this solution to re-implement the former functionality is simpler and ensures players can build in pause mode without as much screen clutter. At a future date the more complex solution may be considered.

### CODE REVIEW

**Comment by janisozaur** [comment]
This requires network version bump

**Comment by KatieZeldaKat** [comment]
Bumped the network version

**Comment by ZehMatt** [comment]
Also add a changelog entry please.

**Comment by KatieZeldaKat** [comment]
Changelog entry added

**Review by ZehMatt — APPROVED** [review]

**Comment by Broxzier** [comment]
I always kinda liked the money effects all moving in sync when you u pause after having built a whole lot.

**Comment by ZehMatt** [comment]
We could turn this into an option also, we can apply the same thing I did to ensure multiplayer stays in sync by simply not drawing them rather than disabling the creation.

**Comment by KatieZeldaKat** [comment]
Just updated the branch to no longer conflict with `develop`.

Is there substantial reason to go with a different solution? Personally, I am fine with the current implementation and would love this change to improve visibility when building in pause mode. Although I can admit to my bias in that.

**Review by ZehMatt — CHANGES_REQUESTED** [review]
Gave it some more thought, this will probably cause desyncs when the host paused the game and performs game actions, since other clients don't have this setting synchronized this would create money entities where the host wouldn't. It might be best to simply not draw the money when paused for clients that have this setting enabled keeping the entities in sync and just hide it visually.

**Comment by KatieZeldaKat** [comment]
Money effects are now always created, and their rendering is what is affected by this change. A new option has been added to the menu which specifies whether building costs should be shown when paused. This is disabled by default:

Options screen showing a new option that says "Show building costs when paused"

[The second implementation listed here](https://github.com/OpenRCT2/OpenRCT2/pull/20796#issuecomment-1728247219) is what I ended up going for in the new changes. It makes it so that in default gameplay, nothing changes in how the game renders the money effects. It's when people enable the cheat for building in pause mode in which people will notice the money effects will no longer clutter the screen.

When resuming the game, the money effects will display, even after pausing the game once again.

File: data/language/en-GB.txt [hunk 1]
```diff
@@ -3697,6 +3697,8 @@ STR_6591    :Staff member is currently fixing a ride and can’t be fired.
 STR_6592    :Staff member is currently inspecting a ride and can’t be fired.
 STR_6593    :Remove park fences
 STR_6594    :Tile Inspector: Toggle wall slope
+STR_6595    :Show building costs when paused
```
Comment (ZehMatt) [inline data/language/en-GB.txt:3700]: I'm not sure if that is the best description, the money effect can appear for removing things as well. I think a better description could be something like "Show money effect when paused"

File: src/openrct2/config/Config.h [hunk 2]
```diff
@@ -58,6 +58,7 @@ struct GeneralConfiguration
     bool RenderWeatherGloom;
     bool DisableLightningEffect;
     bool ShowGuestPurchases;
+    bool ShowBuildCostIfPaused;
```
Comment (ZehMatt) [inline src/openrct2/config/Config.h:61]: I would go for `ShowMoneyEffectIfPaused`, see my comment on the strings.

**Comment by karst** [comment]
I think it makes sense for the animation to play when you unpause the game if the cheat is activated and have the animation not show up yet while paused. You see the money go down. I don't think an option would be necessary for that, it'd just be quality of life improvement for the build in pause mode cheat.

**Comment by ZehMatt** [comment]
Well I'm personally a bit split on this, while its handy to not have the money show up initially it however adds two problems.
1. It will pollute the entity registry with invisible entities basically, if you do a lot of construction while paused it will be relatively easy to have hundreds of even thousands of money entities just wasting away resources.
2. If you did a lot of things while paused and accumulated hundreds if not thousands of money entities it will be quite the show once you unpause the game.

As for the option to just not create the money is not feasible for multiplayer as that will cause desyncs, we had this issue before with the show guest purchases option which I changed to simply not render those when the option is enabled which keeps all clients in sync still.

I think the middle ground is to always animate the money regardless if its paused or not but also additionally have the option to  hide the money as long the game is paused and not just for the time when they are created, that would be also not too difficult to implement without having a ton of new complex logic.

**Comment by KatieZeldaKat** [comment]
Closing in favor of #21102.

It doesn't have the suggested setting to hide all money effects when the game is paused, but I also wouldn't be surprised if it's not needed. I have changed my opinion to say that preserving the functionality of animating all money effects created when paused at the same time when resuming is probably not worth it.

**Pull request closed without merging**